package.json

```diff
@@ -1,66 +1,78 @@
 {
-  "name": "kodj_web_react",
-  "private": true,
-  "version": "0.0.0",
-  "type": "module",
-  "scripts": {
-    "dev": "vite",
-    "build": "tsc -b && vite build",
-    "lint": "eslint .",
-    "format": "prettier --write src/**/*.{ts,tsx}",
-    "preview": "vite preview"
-  },
-  "dependencies": {
-    "@types/react-responsive": "^8.0.8",
-    "@types/react-virtualized": "^9.22.2",
-    "antd": "^5.22.2",
-    "axios": "^1.8.4",
-    "cobe": "^0.6.3",
-    "firebase": "^11.0.2",
-    "firebase-admin": "^13.0.1",
-    "framer-motion": "^11.12.0",
-    "i18next": "^24.0.5",
-    "i18next-browser-languagedetector": "^8.0.4",
-    "lucide-react": "^0.462.0",
-    "prettier": "^3.5.3",
-    "react": "^18.3.1",
-    "react-dom": "^18.3.1",
-    "react-hook-form": "^7.53.2",
-    "react-i18next": "^15.1.3",
-    "react-icons": "^5.3.0",
-    "react-responsive": "^10.0.1",
-    "react-router-dom": "^6.28.0",
-    "react-virtualized": "^9.22.6",
-    "recharts": "^2.15.0",
-    "recoil": "^0.7.7",
-    "rollup-plugin-visualizer": "^5.13.1",
-    "styled-components": "^6.1.14"
-  },
-  "devDependencies": {
-    "@eslint/js": "^9.15.0",
-    "@types/node": "^22.10.1",
-    "@types/react": "^18.3.12",
-    "@types/react-dom": "^18.3.1",
-    "@vitejs/plugin-react": "^4.3.4",
-    "eslint": "^9.15.0",
-    "eslint-plugin-react-hooks": "^5.0.0",
-    "eslint-plugin-react-refresh": "^0.4.14",
-    "globals": "^15.12.0",
-    "typescript": "~5.6.2",
-    "typescript-eslint": "^8.15.0",
-    "vite": "^6.0.1"
-  },
-  "description": "<div align=\"center\" style=\"width: 100%\">   <img src=\"https://github.com/user-attachments/assets/75e522ad-09c9-4d2e-99b4-bd6f6ea158b7\" alt=\"kodj_logo\" height=\"600\" width=\"auto\" /> </div> &nbsp; <div align=center >    <img src=\"https://img.shields.io/badge/React-61DAFB?style=flat-square&logo=React&logoColor=black\"/>    <img src=\"https://img.shields.io/badge/Typescript-3178C6?style=flat-square&logo=Typescript&logoColor=white\"/> </div>",
-  "main": "eslint.config.js",
-  "repository": {
-    "type": "git",
-    "url": "git+https://github.com/Sardor-M/kodj_web_react.git"
-  },
-  "keywords": [],
-  "author": "",
-  "license": "ISC",
-  "bugs": {
-    "url": "https://github.com/Sardor-M/kodj_web_react/issues"
-  },
-  "homepage": "https://github.com/Sardor-M/kodj_web_react#readme"
+    "name": "kodj_web_react",
+    "private": true,
+    "version": "0.0.0",
+    "type": "module",
+    "scripts": {
+        "dev": "vite",
+        "build": "tsc -b && vite build",
+        "lint": "eslint .",
+        "format": "prettier --write src/**/*.{ts,tsx}",
+        "preview": "vite preview",
+        "prepare": "husky"
+    },
+    "lint-staged": {
```

Comment:
Added ` Husky` and lint-staged configuration to enforce code quality standards before commits. This setup automatically:
- Runs ESLint with auto-fix on staged TypeScript files
- Formats all staged TS/TSX files with Prettier
- Formats staged JSON and Markdown files with Prettier

This pre-commit hook runs only on staged files before commit and automatically validates and formats code, ensuring consistent code style across the project without requiring manual checks before each commit.

Note: Manual `lint` and `format` scripts remain available for on-demand checks of the entire codebase when needed.

Having control over checking/formatting files without having to stage them, while automating the process for only the changed files, gives us more flexibility and freedom.